### DISCUSSION TITLE
Extend / inherit from existing plugins in Parcel 2

### DISCUSSION THREAD

**Discussion by AndyOGo** [parcel-bundler/parcel#7680] [Q&A]
In Parcel V1 it was easy to inherit from an existing [`AssetType`](https://v1.parceljs.org/asset_types.html) in [`parcel/src/assets`](https://github.com/parcel-bundler/parcel/tree/master/packages/core/parcel-bundler/src/assets) and just assign the new class.

To speak in code:

```js
const JSAsset = require('parcel-bundler/src/assets/JSAsset')

module.exports = class MyJSAsset extends JSAsset {
  // my JS asset customization here
}
```

```js
const Bundler = require('parcel-bundler')

let bundler = new Bundler('input.js')
bundler.addAssetType('.js', require.resolve('./MyJSAsset'))
```

But with Parcel V2 modules return instances, not classes.
Is it supposed to extend existing plugins via mixins?
That would need a config file change to remove the extended plugin too, right?

**Comment by AndyOGo** [comment]
@devongovett 
@mischnic 
Can you point me to a direction?

**Reply by mischnic** [reply]
I think there's no easy way to do this right now...

> That would need a config file change to remove the extended plugin too, right?

Parcel 1 had just 1 plugin for each file type, now you can have multiple plugins that are executed in order. This is a sideeffect of that

**Reply by AndyOGo** [reply]
thanks @mischnic 

So I assume, doing something like this isn't working as expected?

```js
import { Transformer } from "@parcel/plugin";
import JSTransformer from "@parcel/transformer-js";

export default Transformer({
  ...JSTransformer,
  async transform(opts) {
    // do my custom transform stuff here
    
    return await JSTransformer.transform(opts);
  },
})
```

Then I would create my own `.parcelrc` file overriding this

https://github.com/parcel-bundler/parcel/blob/e0440dcbe6d6e419a412c8c568d94715cdb4282a/packages/configs/default/index.json#L11-L15
